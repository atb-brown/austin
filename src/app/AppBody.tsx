import AppLinks from "./AppLinks";
import AppWidgets from "./AppWidgets";
import React from "react";

/**
 * The primary body of the site that I want prominently displayed on my site.
 *
 * @return {ReactElement}
 */
export default function AppBody() {
  return (
    <div style={{ display: "flex", width: "100%" }}>
      <div style={{ display: "flex", flex: 1 }}>
        <AppLinks />
      </div>
      <div style={{ display: "flex", flex: 3 }}>
        <AppWidgets />
      </div>
    </div>
  );
}
