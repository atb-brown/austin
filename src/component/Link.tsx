import React from "react";

/**
 * A basic, clickable link for a URL.
 *
 * @param {LinkProps} props Properties.
 * @return {ReactElement} The link component.
 */
export default function Link({ description, url }: LinkProps): JSX.Element {
  return (
    <a className="App-link" href={url}>
      {description}
    </a>
  );
}

type LinkProps = {
  readonly description: string;
  readonly url: string;
};
