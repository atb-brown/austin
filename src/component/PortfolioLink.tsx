import Link from "./Link";
import React from "react";

export default function PortfolioLink(props: LinkProps) {
  if (props.issuerDescription && props.issuerUrl) {
    return (
      <LinkWithIssuer
        description={props.description}
        url={props.url}
        urlId={props.urlId}
        issuerDescription={props.issuerDescription}
        issuerUrl={props.issuerUrl}
      />
    );
  }
  return (
    <LinkWithoutIssuer
      description={props.description}
      url={props.url}
      urlId={props.urlId}
    />
  );
}

function LinkWithoutIssuer(props: LinkProps) {
  // TODO: it's a bad idea to have <div/> elements that are children of <p/> elements
  return (
    <div>
      <div>{props.description}</div>
      <div>
        <Link description={props.urlId} url={props.url} />
      </div>
      <br />
    </div>
  );
}

function LinkWithIssuer(props: LinkProps) {
  return (
    <div>
      <div>{props.description}</div>
      <div>
        <Link description={props.urlId} url={props.url} />
      </div>
      <div className="App-soft-text">
        Issued by:{" "}
        <Link
          description={props.issuerDescription ?? ""}
          url={props.issuerUrl ?? ""}
        />
      </div>
      <br />
    </div>
  );
}

type LinkProps = {
  description: string;
  url: string;
  urlId: string;
  issuerDescription?: string;
  issuerUrl?: string;
};
