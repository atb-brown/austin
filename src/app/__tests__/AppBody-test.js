import { wait as guestBook } from "../../test/component/GuestBookTestUtils";
import { mockProvider as mockGuestBookInfoProvider } from "../../test/mock/hook/mockUseGuestBookInfo";
import AppBody from "../AppBody";
import { render } from "@testing-library/react";

it("renders to match snapshot", async () => {
  mockGuestBookInfoProvider();

  const renderResult = render(<AppBody />);

  await guestBook(renderResult);

  expect(renderResult.asFragment()).toMatchSnapshot();
});

afterEach(() => {
  jest.resetAllMocks();
});
