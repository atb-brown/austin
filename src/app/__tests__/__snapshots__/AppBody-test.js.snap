// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to match snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: flex; width: 100%;"
  >
    <div
      style="display: flex; flex: 1;"
    >
      <div
        style="display: flex; flex: 1; flex-direction: column; align-items: center;"
      >
        <h1>
          Links
        </h1>
        <div
          class="App-text"
        >
          <div>
            GitHub
          </div>
          <div>
            <a
              class="App-link"
              href="https://github.com/atb-brown"
            >
              atb-brown
            </a>
          </div>
          <br />
        </div>
        <div
          class="App-text"
        >
          <div>
            StackOverflow
          </div>
          <div>
            <a
              class="App-link"
              href="https://stackoverflow.com/users/5476186/austin-brown"
            >
              5476186/austin-brown
            </a>
          </div>
          <br />
        </div>
        <div
          class="App-text"
        >
          <div>
            LinkedIn
          </div>
          <div>
            <a
              class="App-link"
              href="https://www.linkedin.com/in/austin-brown-06382a12a/"
            >
              austin-brown-06382a12a
            </a>
          </div>
          <br />
        </div>
        <div
          class="App-text"
        >
          <div>
            Scrum Master Certification
          </div>
          <div>
            <a
              class="App-link"
              href="https://bcert.me/sixkczuof"
            >
              sixkczuof
            </a>
          </div>
          <div
            class="App-soft-text"
          >
            Issued by: 
            <a
              class="App-link"
              href="https://www.scrumalliance.org/"
            >
              Scrum Alliance, Inc.
            </a>
          </div>
          <br />
        </div>
        <div
          class="App-text"
        >
          <div>
            SAFe 5 Scrum Master Certification
          </div>
          <div>
            <a
              class="App-link"
              href="https://www.credly.com/badges/18e3c923-956c-4525-b857-5f9c0dfb9713/public_url"
            >
              18e3c923-956c-4525-b857-5f9c0dfb9713
            </a>
          </div>
          <div
            class="App-soft-text"
          >
            Issued by: 
            <a
              class="App-link"
              href="https://scaledagile.com/"
            >
              Scaled Agile, Inc.
            </a>
          </div>
          <br />
        </div>
      </div>
    </div>
    <div
      style="display: flex; flex: 3;"
    >
      <div
        style="width: 100%; flex: 1; display: flex; flex-direction: column; align-items: center; justify-content: center; gap: 10px;"
      >
        <div
          style="width: 100%; flex: 1; display: flex; flex-direction: column; align-items: center; justify-content: center;"
        >
          <div
            class="disable-text-selection"
            data-testid="counter"
            style="display: flex; font-size: 20px; background-color: steelblue; align-items: center; justify-content: center; width: 25%; height: 60px; border-radius: 10px;"
            title="This component counts the number of times it has been clicked. It's an exercise in handling clicks and component state!"
          >
            0
          </div>
        </div>
        <div
          style="width: 100%; flex: 8; display: flex; flex-direction: column; align-items: center; justify-content: center;"
        >
          <div
            data-testid="guest-book"
            style="display: flex; min-height: 50%; width: 99%; flex-direction: column; gap: 3px;"
            title="This component displays everyone who has signed my guest book. It's an exercise in HTTP requests and dependency injection!"
          >
            <div
              style="border-top-left-radius: 15px; border-top-right-radius: 15px; border-bottom-left-radius: 15px; border-bottom-right-radius: 15px; overflow: hidden; display: flex; width: 95%; height: 85px;"
            >
              <div
                class="Guests-Visual"
                style="flex: 2; flex-direction: column; justify-content: space-evenly;"
              >
                <div
                  class="Guests-Text-container"
                  style="font-size: 75%;"
                >
                  austin
                </div>
                <div
                  class="Guests-Text-container"
                  style="font-size: 50%;"
                >
                  www.example.com
                </div>
              </div>
              <div
                class="Guests-Visual Guests-Text-Container"
                style="flex: 7; margin-left: 3px; margin-right: 3px;"
              >
                <div
                  class="Guests-Text-Container"
                  style="font-size: 75%; height: 100%; width: 98%;"
                >
                  info loaded
                </div>
              </div>
              <div
                class="Guests-Visual Guests-Text-Container"
                style="flex: 1; font-size: 60%;"
              >
                2023
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
